### PR TITLE
Update python 3.8 and 3.10

### PR DESCRIPTION
**Pull request by hgy59**
## Description

Update Python 3.10.4 to v3.10.5 
Update Python 3.8.12 to v3.8.13

Fixes # 

## Checklist

- [ ] Build rule `all-supported` completed successfully
- [ ] New installation of package completed successfully
- [ ] Package upgrade completed successfully (Manually install the package again)
- [ ] Package [functionality was tested](https://github.com/SynoCommunity/spksrc/wiki/Update-Policy#tests-checks)
- [ ] Any needed [documentation](https://github.com/SynoCommunity/spksrc/wiki/Create-documentation) is updated/created

### Type of change

- [ ] Bug fix
- [ ] New Package
- [x] Package update
- [ ] Includes small framework changes
- [ ] This change requires a documentation update (e.g. Wiki)

### CODE REVIEW

**Comment by Safihre** [comment]
@hgy59 It seems `domoticz` is the last `python38` package, is there a reason it can't simply be switches to `python310`?
Then we can also get rid of the special `cross/setuptools` and `cross/wheel` cases.
Also, do we need `cross/pip` with all the other stuff that we do with `get-pip` and `crossenv`?

**Comment by hgy59** [comment]
> @hgy59 It seems `domoticz` is the last `python38` package, is there a reason it can't simply be switches to `python310`?

Probably you mean `deluge` as domoticz does not use python. Deluge is WIP in  #4153.
Latest published versions of `sickchill` and `homeassistant` are still Python 3.8.

To update homeassistant, I am still waiting for a solution to update cryptography with dependency of setuptools_rust.

File: cross/python310/Makefile [hunk 1]
```diff
@@ -133,24 +140,10 @@ python310_post_install: $(WORK_DIR)/python-cc.mk
...
-	. $(WORK_DIR)/crossenv/bin/activate && $(RUN) pip install "poetry==1.1.13"
-	. $(WORK_DIR)/crossenv/bin/activate && $(RUN) build-pip install "Cython==0.29.28"
-	. $(WORK_DIR)/crossenv/bin/activate && $(RUN) pip install "Cython==0.29.28"
-	. $(WORK_DIR)/crossenv/bin/activate && $(RUN) build-pip install "flit==3.7.1"
-	. $(WORK_DIR)/crossenv/bin/activate && $(RUN) pip install "flit==3.7.1"
+	. $(WORK_DIR)/crossenv/bin/activate && $(RUN) build-python get-pip.py "pip==22.0.4" --no-setuptools --no-wheel --disable-pip-version-check
+	. $(WORK_DIR)/crossenv/bin/activate && $(RUN) build-pip --disable-pip-version-check install "setuptools==62.4.0" "setuptools-scm==6.4.2" "wheel==0.37.1" "cffi==1.15.0" "cryptography==3.3.2" "poetry==1.1.13" "Cython==0.29.28" "flit==3.7.1"
+	. $(WORK_DIR)/crossenv/bin/activate && $(RUN) python get-pip.py "pip==22.0.4" --no-setuptools --no-wheel --disable-pip-version-check
+	. $(WORK_DIR)/crossenv/bin/activate && $(RUN) pip --disable-pip-version-check install "setuptools==62.4.0" "setuptools-scm==6.4.2" "wheel==0.37.1" "cffi==1.15.0" "cryptography==3.3.2" "poetry==1.1.13" "Cython==0.29.28" "flit==3.7.1"
```
Comment (th0ma7) [inline cross/python310/Makefile:146]: I like that you kept `--no-setuptools --no-wheel` that @Safihre used.  This gives a more granular management of versions as I did encountered issues in the past.

**Review by th0ma7 — APPROVED** [review]

**Comment by th0ma7** [comment]
python 3.10.5 was on my todo list, thnx for handling that!

As for cross `cffi`, `greenlet`, `pycparser`, `setuptools` and `wheel` (and sub-dependent `cryptography` and all) are now only used for python38 now (and perhaps legacy python2).  In theory they could all be removed in favor of using `crossenv`|`pure` instead thus removing tons of unecessary code to maintain.

That's the strategy I used for python310 along with reduing it to its bare minimal set of wheel.

**Comment by hgy59** [comment]
@th0ma7 ffsync does not build anymore

Probably it has to be adjusted for redesigned wheel building?

```
.PHONY: ffsync_extra_install
ffsync_extra_install:
	install -m 755 -d $(STAGING_DIR)/var
	install -m 644 src/ffsync.ini $(STAGING_DIR)/var/
	sed -i -e "s|https://github.com/mozilla-services/mozservices/archive/e00e1b68130423ad98d0f6185655bde650443da8.zip|mozsvc==0.8|g" \
	       -e "s|https://github.com/mozilla-services/tokenserver/archive/d7e513e8a4f5c588b70d685a8df1d2e508c341c0.zip|tokenserver==1.2.7|g" \
	       -e "s|https://github.com/mozilla-services/server-syncstorage/archive/1.5.5.zip|SyncStorage==1.5.5|g" \
	       -e "s|https://github.com/mozilla-services/syncserver/archive/1.5.2.tar.gz|syncserver==1.5.2|g" \
                $(STAGING_DIR)/share/wheelhouse/requirements.txt
```

requirements.txt contains:

```
https://github.com/mozilla-services/tokenserver/archive/1.5.11.tar.gz
https://github.com/mozilla-services/server-syncstorage/archive/1.8.0.tar.gz
https://github.com/mozilla-services/syncserver/archive/1.9.1.tar.gz
```

it fails with:

```
 make[3]: Leaving directory '/github/workspace/spk/ffsync'
===>  [https://github.com/mozilla-services/tokenserver/archive/1.5.11.tar.gz]
make[3]: Entering directory '/github/workspace/spk/ffsync'
make[3]: *** No rule to make target 'cross-compile-wheel-https://github.com/mozilla-services/tokenserver/archive/1.5.11.tar.gz'.  Stop.
make[3]: Leaving directory '/github/workspace/spk/ffsync'
make[2]: *** [../../mk/spksrc.wheel.mk:81: build_wheel_target] Error 1
make[2]: Leaving directory '/github/workspace/spk/ffsync'
```

**Comment by Safihre** [comment]
@hgy59 confusing, because only vim and domoticz still mention `python38`. I didn't check the published ones, assumed they would match the repo here.

**Comment by th0ma7** [comment]
> @th0ma7 ffsync does not build anymore
> 
> Probably it has to be adjusted for redesigned wheel building?
> 
> ```
> .PHONY: ffsync_extra_install
> ffsync_extra_install:
> 	install -m 755 -d $(STAGING_DIR)/var
> 	install -m 644 src/ffsync.ini $(STAGING_DIR)/var/
> 	sed -i -e "s|https://github.com/mozilla-services/mozservices/archive/e00e1b68130423ad98d0f6185655bde650443da8.zip|mozsvc==0.8|g" \
> 	       -e "s|https://github.com/mozilla-services/tokenserver/archive/d7e513e8a4f5c588b70d685a8df1d2e508c341c0.zip|tokenserver==1.2.7|g" \
> 	       -e "s|https://github.com/mozilla-services/server-syncstorage/archive/1.5.5.zip|SyncStorage==1.5.5|g" \
> 	       -e "s|https://github.com/mozilla-services/syncserver/archive/1.5.2.tar.gz|syncserver==1.5.2|g" \
>                 $(STAGING_DIR)/share/wheelhouse/requirements.txt
> ```
> 
> requirements.txt contains:
> 
> ```
> https://github.com/mozilla-services/tokenserver/archive/1.5.11.tar.gz
> https://github.com/mozilla-services/server-syncstorage/archive/1.8.0.tar.gz
> https://github.com/mozilla-services/syncserver/archive/1.9.1.tar.gz
> ```
> 
> it fails with:
> 
> ```
>  make[3]: Leaving directory '/github/workspace/spk/ffsync'
> ===>  [https://github.com/mozilla-services/tokenserver/archive/1.5.11.tar.gz]
> make[3]: Entering directory '/github/workspace/spk/ffsync'
> make[3]: *** No rule to make target 'cross-compile-wheel-https://github.com/mozilla-services/tokenserver/archive/1.5.11.tar.gz'.  Stop.
> make[3]: Leaving directory '/github/workspace/spk/ffsync'
> make[2]: *** [../../mk/spksrc.wheel.mk:81: build_wheel_target] Error 1
> make[2]: Leaving directory '/github/workspace/spk/ffsync'
> ```

Most probably... I recall finding some other sources for that when I tried to migrate it to python3 (which is literally not possible).  Got a little too much on my TODO list but I'll come back to it eventually.  We had mentioned of revisiting the python2 package for ffsync.  Something else I'd like to get into once python3 migration to py310 is over.

**Comment by Safihre** [comment]
@hgy59 should we merge this?

**Pull request merged**